dashboard/test/ui/features/step_definitions/google_blockly.rb

```diff
@@ -0,0 +1,28 @@
+# Google Blockly blocks do not have hard-coded IDs. Blocks are specified in document order.
+
+# Note: this is an offset relative to the current position of the block
+When /^I drag Google Blockly block "([^"]*)" to offset "([^"]*), ([^"]*)"$/ do |block_index, dx, dy|
+  @browser.execute_script("$(\"[class='blocklyDraggable']\").eq(#{block_index.to_i}).simulate( 'drag', {handle: 'corner', dx: #{dx}, dy: #{dy}, moves: 5});")
+end
+
+When /^I drag Google Blockly block "([^"]*)" to block "([^"]*)"$/ do |target_index, destination_index|
+  code = generate_google_blockly_block_drag_code(target_index, destination_index, 0, 30)
+  @browser.execute_script code
+end
+
+Then /^Google Blockly block "([^"]*)" is near offset "([^"]*), ([^"]*)"$/ do |block_index, x, y|
+  point = get_google_blockly_block_coordinates(block_index)
+  expect(point.x).to be_within(3).of(x.to_i)
+  expect(point.y).to be_within(3).of(y.to_i)
+end
+
+Then /^Google Blockly block "([^"]*)" is child of block "([^"]*)"$/ do |child_index, parent_index|
+  id_selector = get_id_selector
+  @child_item = @browser.find_elements(class: 'blocklyDraggable')[child_index.to_i]
```

Comment:
do we want these variables (`@child_item`, `@parent_item`, etc) to be instance variables ([here's more about instance variables in ruby](https://www.rubyguides.com/2019/07/ruby-instance-variables/))? that implies that these variables are accessed outside the scope of this "Then" block, which is seems unlikely

Reply:
Thanks for always including links like this to help me learn!
I've tested and confirmed that this step definition works fine without the use of instance variables. I chose to use them initially because I was following what I perceived as conventions being set here:
https://github.com/code-dot-org/code-dot-org/blob/6bd5260de0002935f3cbb3604e6d6f4d92bd1ed5/dashboard/test/ui/features/step_definitions/blockly.rb#L116-L123